client/http/client.go

```diff
@@ -289,23 +293,57 @@ func (c *client) GetHotReadRegions(ctx context.Context) (*StoreHotPeersInfos, er
 // GetHotWriteRegions gets the hot write region statistics info.
 func (c *client) GetHotWriteRegions(ctx context.Context) (*StoreHotPeersInfos, error) {
 	var hotWriteRegions StoreHotPeersInfos
-	err := c.requestWithRetry(ctx, "GetHotWriteRegions", HotWrite, &hotWriteRegions)
+	err := c.requestWithRetry(ctx, "GetHotWriteRegions", HotWrite, http.MethodGet, &hotWriteRegions)
 	if err != nil {
 		return nil, err
 	}
 	return &hotWriteRegions, nil
 }
 
+// GetRegionStatusByKey gets the region status by key range.
+func (c *client) GetRegionStatusByKey(ctx context.Context, startKey, endKey []byte) (*RegionStats, error) {
+	var regionStats RegionStats
+	err := c.requestWithRetry(
+		ctx, "GetRegionStatusByKey",
+		RegionStatsByStartEndKey(startKey, endKey), http.MethodGet,
+		&regionStats,
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &regionStats, nil
+}
+
 // GetStores gets the stores info.
 func (c *client) GetStores(ctx context.Context) (*StoresInfo, error) {
 	var stores StoresInfo
-	err := c.requestWithRetry(ctx, "GetStores", Stores, &stores)
+	err := c.requestWithRetry(ctx, "GetStores", Stores, http.MethodGet, &stores)
 	if err != nil {
 		return nil, err
 	}
 	return &stores, nil
 }
 
+// GetPlacementRulesByGroup gets the placement rules by group.
+func (c *client) GetPlacementRulesByGroup(ctx context.Context, group string) ([]*Rule, error) {
+	var rules []*Rule
+	err := c.requestWithRetry(ctx, "GetPlacementRulesByGroup", PlacementRulesByGroup(group), http.MethodGet, &rules)
+	if err != nil {
+		return nil, err
+	}
+	return rules, nil
+}
+
+// SetPlacementRule sets the placement rule.
+func (c *client) SetPlacementRule(ctx context.Context, rule *Rule) error {
+	return c.requestWithRetry(ctx, "SetPlacementRule", PlacementRule, http.MethodPost, nil)
```

Comment:
Does this function seem to be invalid without using `rule`?

Reply:
Nice catch. I think I should also add some tests.